### PR TITLE
add metrics enable switch

### PR DESCRIPTION
**Pull request by finefuture**
## What is the purpose of the change

If user use custom metrics component, we need support switch for default MetricsFilter.
The default value of enabled is true.

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/filter/MetricsFilter.java [hunk 1]
```diff
@@ -40,20 +41,24 @@
...
 
     private ApplicationModel applicationModel;
-    private final static ErrorTypeAwareLogger LOGGER = LoggerFactory.getErrorTypeAwareLogger(MetricsFilter.class);
+    private static final ErrorTypeAwareLogger LOGGER = LoggerFactory.getErrorTypeAwareLogger(MetricsFilter.class);
+    private boolean metricsEnable;
 
     @Override
     public void setApplicationModel(ApplicationModel applicationModel) {
         this.applicationModel = applicationModel;
+        this.metricsEnable = applicationModel.getApplicationConfigManager().getMetrics().map(MetricsConfig::getEnabled).orElse(false);
```
Comment (AlbumenJ) [inline dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/filter/MetricsFilter.java:50]:
```suggestion
        this.metricsEnable = applicationModel.getApplicationConfigManager().getMetrics().map(MetricsConfig::getEnabled).orElse(true);
```

Enabling metrics by default can provide users with a better experience.
BTW, if micrometer cannot found, Dubbo will disable metrics.

Comment (finefuture) [reply dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/filter/MetricsFilter.java]: Fixed, i think we need add some tips on user page, like : dubbo.metrics.enabled = true  to use default metrics

File: dubbo-config/dubbo-config-spring/src/main/resources/META-INF/dubbo.xsd [hunk 2]
```diff
@@ -1098,6 +1098,12 @@
                 <xsd:documentation><![CDATA[ Deprecated. No longer use. ]]></xsd:documentation>
             </xsd:annotation>
         </xsd:attribute>
+
```
Comment (songxiaosheng) [inline dubbo-config/dubbo-config-spring/src/main/resources/META-INF/dubbo.xsd:1101]: 开关可以默认是true

**Review by songxiaosheng — CHANGES_REQUESTED** [review]

File: dubbo-config/dubbo-config-spring/src/main/resources/META-INF/dubbo.xsd (hunk 2 above)
Comment (finefuture) [reply dubbo-config/dubbo-config-spring/src/main/resources/META-INF/dubbo.xsd:1101]: Fixed

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/filter/MetricsFilter.java [hunk 3]
```diff
@@ -40,20 +41,24 @@
...
     }
 
     @Override
     public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcException {
-        try {
-            RequestEvent requestEvent = RequestEvent.toRequestEvent(applicationModel, invocation);
-            MetricsEventBus.before(requestEvent, () -> invocation.put(METRIC_FILTER_EVENT, requestEvent));
-        } catch (Throwable t) {
-            LOGGER.warn(INTERNAL_ERROR, "", "", "Error occurred when invoke.", t);
+        if (metricsEnable) {
```
Comment (songxiaosheng) [inline dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/filter/MetricsFilter.java:55]: 这个开关要不要改下名字，这个只是配置RPC的开关，我们指标还有三大中心、线程池、应用等指标信息，这个开关可以只控制RPC的指标采集，可以把名字改下见名知意

**Review by songxiaosheng — CHANGES_REQUESTED** [review]

File: dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/filter/MetricsFilter.java (hunk 3 above)
Comment (finefuture) [reply dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/filter/MetricsFilter.java]: 改成enableRpc了

File: dubbo-config/dubbo-config-api/src/test/java/org/apache/dubbo/config/AbstractConfigTest.java [hunk 4]
```diff
@@ -1002,7 +1002,7 @@ void testMetaData() throws Exception {
         List<Class<? extends AbstractConfig>> configClasses = Arrays.asList(ApplicationConfig.class,
                 ConsumerConfig.class, ProviderConfig.class, ReferenceConfig.class, ServiceConfig.class,
                 ProtocolConfig.class, RegistryConfig.class, ConfigCenterConfig.class, MetadataReportConfig.class,
-                ModuleConfig.class, SslConfig.class, MetricsConfig.class, MonitorConfig.class, MethodConfig.class);
+                ModuleConfig.class, SslConfig.class, MonitorConfig.class, MethodConfig.class);
```
Comment (AlbumenJ) [inline dubbo-config/dubbo-config-api/src/test/java/org/apache/dubbo/config/AbstractConfigTest.java:1005]: Why change this?

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-config/dubbo-config-api/src/test/java/org/apache/dubbo/config/AbstractConfigTest.java (hunk 4 above)
Comment (AlbumenJ) [reply dubbo-config/dubbo-config-api/src/test/java/org/apache/dubbo/config/AbstractConfigTest.java:1005]: Is there any problem if set null as default?

Comment (finefuture) [reply dubbo-config/dubbo-config-api/src/test/java/org/apache/dubbo/config/AbstractConfigTest.java]: Fixed

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**